ui/hooks/usePolling.ts

```diff
@@ -13,13 +13,20 @@ type UsePollingOptions = {
   // TODO: Replace `any` with type
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   options?: any;
+  enabled?: boolean;
 };
 
 const usePolling = (usePollingOptions: UsePollingOptions) => {
   const pollTokenRef = useRef<null | string>(null);
   const cleanupRef = useRef<null | ((pollingToken: string) => void)>(null);
   let isMounted = false;
   useEffect(() => {
+    if (usePollingOptions.enabled === false) {
+      return () => {
+        // noop
+      };
```

Comment:
why is this a noop function? Is this just overriding the cleanup? Or I guess is it overriding everything below?

Reply:
I guess that makes sense, just not familiar with this pattern

Reply:
useEffect expects undefined or a cleanup fn, typescript wasn't happy with mixing returning undefined here and the existing cleanup fn below, so I use a noop fn instead